shell/platform/android/image_external_texture_gl.cc

```diff
@@ -64,12 +55,37 @@ bool ImageExternalTextureGL::MaybeSwapImages() {
   JavaLocalRef old_android_image(android_image_);
   android_image_.Reset(image);
   JavaLocalRef hardware_buffer = HardwareBufferFor(image);
-  egl_image_ = CreateEGLImage(AHardwareBufferFor(hardware_buffer));
+  AHardwareBuffer* latest_hardware_buffer = AHardwareBufferFor(hardware_buffer);
+  HardwareBufferKey key =
+      flutter::NDKHelpers::AHardwareBuffer_getId(latest_hardware_buffer);
+  auto existing_image = image_lru_.FindImage(key);
+  if (existing_image != nullptr) {
+    dl_image_ = existing_image;
+
+    CloseHardwareBuffer(hardware_buffer);
+    // IMPORTANT: We have just received a new frame to display so close the
+    // previous Java Image so that it is recycled and used for a future frame.
+    CloseImage(old_android_image);
+    return;
+  }
+
+  auto egl_image = CreateEGLImage(latest_hardware_buffer);
   CloseHardwareBuffer(hardware_buffer);
-  // IMPORTANT: We only close the old image after egl_image_ stops referencing
-  // it.
+  // IMPORTANT: We have just received a new frame to display so close the
+  // previous Java Image so that it is recycled and used for a future frame.
   CloseImage(old_android_image);
-  return true;
+
+  if (!egl_image.is_valid()) {
+    return;
+  }
+
+  dl_image_ = CreateDlImage(context, bounds, key, std::move(egl_image));
+  gl_entry_.erase(image_lru_.AddImage(dl_image_, key));
+}
+
+void ImageExternalTextureGL::Detach() {
+  // TODO: should we clear the cache in detach?
```

Comment:
Somehow my comment got lost, but, yes, we clear gl_entry_ here and the code relies on gl_entry_ being kept in sync with the cache. By clearing gl_entry_ here you actually invalidate the backing eglImages and textures referenced by the cache.

Reply:
I assume that we also be clearing the cache in the vk implementation too?

Reply:
But I guess we don't have anything in the vk cache besides the imported hardware buffer textures

Reply:
I think we probably want to clear the cache in vulkan too.